proto-lens-tests/tests/decode_delimited.hs

```diff
@@ -0,0 +1,41 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Main (main) where
+
+import qualified Data.ByteString as B
+import Data.ProtoLens
+import Data.ProtoLens.Encoding.Bytes (runBuilder)
+import Lens.Family2 ((&), (.~))
+import Test.Framework.Providers.HUnit (testCase)
+import Test.HUnit ((@=?))
+
+import Data.ProtoLens.TestUtil
+import Proto.DecodeDelimited
+import Proto.DecodeDelimited_Fields
+
+import System.IO (openBinaryFile, hClose, IOMode(ReadMode))
+import System.Directory (removeFile)
+import Control.Exception (catch, throwIO)
+import System.IO.Error (isDoesNotExistError)
+
+filename :: String
+filename = "test_decode_delimited.dat"
+
+removeIfExists :: FilePath -> IO ()
+removeIfExists fileName = removeFile fileName `catch` handleExists
+  where handleExists e
+          | isDoesNotExistError e = return ()
+          | otherwise = throwIO e
+
+main :: IO ()
+main = testMain
+    [ testCase "buildDelimited/decodeDelimited" $ do
+          let foo = defMessage & a .~ 42 & b .~ "hello" :: Foo
+          let bs = runBuilder . buildMessageDelimited $ foo
+          B.writeFile filename bs
+          h <- openBinaryFile filename ReadMode
+          any1 <- decodeDelimitedMessageH h :: IO (Either String Foo)
+          hClose h
+          removeIfExists filename
```

Comment:
When would we not expect the file to exist?  Would it make the test simpler/more robust if we just threw an error if the file unintentionally disappeared?

Reply:
That makes sense. Do you want me to leave just the call to `removeFile` (which might throw) without additional bookkeeping.

Reply:
`withSystemTempFile` should remove the file automatically, so I believe you could just delete the call.